---
title: 'PGS-220000 PostgreSQL error: relation "t1" does not exist'
description: Description
solution: Campaign Classic v7
product: Campaign Classic v7
applies-to: Campaign Classic v7
keywords: KCS, Campaign Classic, Tracking workflow, nms:TrackingLog
resolution: Resolution
internal-notes: null
bug: false
article-created-by: Amol Mahajan
article-created-date: 12/12/2022 1:58:06 PM
article-published-by: Amol Mahajan
article-published-date: 12/12/2022 2:12:42 PM
version-number: 3
article-number: KA-17530
dynamics-url: https://adobe-ent.crm.dynamics.com/main.aspx?forceUCI=1&pagetype=entityrecord&etn=knowledgearticle&id=35fe2fff-247a-ed11-81ac-6045bd0063aa
exl-id: 1a8a9bfd-7dcf-4a7d-b87e-0f17e6e63a08
---
# PGS-220000 PostgreSQL error: relation "t1" does not exist

## Description {#description}

<b>Environment</b>
Campaign Classic V7


<b>Issue/Symptoms</b>
After a recent build upgrade 9343 or higher, the following errors are generated by the tracking workflow:




```
nlmodule    PGS-220000 PostgreSQL error: ERROR:  relation "t1" does not exist LINE 1: ...t, T3.iArticle, T2.iType FROM NmsTrackingUrl T2,  T1 LEFT OU... nlmodule    WDB-200001 SQL statement 'SELECT T1.tsLog, T1.iDeliveryId, T1.iUrlId, T1.sSourceType, T3.dAmount, T3.iArticle, T2.iType FROM NmsTrackingUrl T2,  T1 LEFT OUTER JOIN NmsWebTrackingLog T3 ON iTrackingLogId = iWebTrackingLogId WHERE T1.tsLog  :#(1)# AND T1.tsLog = :#(2)#  AND T1.iUrlId=T2.iTrackingUrlId ORDER BY T1.tsLog' could not be executed.   Param(0)=09/17/2020 8:55:00 PM   Param(1)=09/16/2020 8:55:00 PMnlmodule    Attribute 'broadLog-id' unknown (see definition of schema 'Tracking logs (nms:trackingLog)').
```





## Resolution {#resolution}


From the verbose tracking logs, one can see that the tracking workflow is failing at the end, of consolidating the data:




```
00000000 00000000 1 info log Consolidating tracking logs for the period 09/16/2020 3:55:00 PM, 09/17/2020 3:55:00 PM...
00000000 00000000 1 info wdbc Select: 00000000  SELECT T1.tsLog, T1.iDeliveryId, T1.iUrlId, T1.sSourceType, T3.dAmount, T3.iArticle, T2.iType FROM NmsTrackingUrl T2, T1 LEFT OUTER JOIN NmsWebTrackingLog T3 ON iTrackingLogId = iWebTrackingLogId WHERE T1.tsLog  :#(1)# AND T1.tsLog = :#(2)# AND T1.iUrlId=T2.iTrackingUrlId ORDER BY T1.tsLog
00000000 00000000 2 info wdbc Param(0)=09/17/2020 8:55:00 PM
00000000 00000000 2 info wdbc Param(1)=09/16/2020 8:55:00 PM
00000000 00000000 1 error log PGS-220000 PostgreSQL error: ERROR: relation "t1" does not exist\nLINE 1: ...t, T3.iArticle, T2.iType FROM NmsTrackingUrl T2, T1 LEFT OU...\n \n. (iRc=-2006)
00000000 00000000 1 error log WDB-200001 SQL statement 'SELECT T1.tsLog, T1.iDeliveryId, T1.iUrlId, T1.sSourceType, T3.dAmount, T3.iArticle, T2.iType FROM NmsTrackingUrl T2, T1 LEFT OUTER JOIN NmsWebTrackingLog T3 ON iTrackingLogId = iWebTrackingLogId WHERE T1.tsLog  :#(1)# AND T1.tsLog = :#(2)# AND T1.iUrlId=T2.iTrackingUrlId ORDER BY T1.tsLog' could not be executed.\n Param(0)=09/17/2020 8:55:00 PM\n Param(1)=09/16/2020 8:55:00 PM (iRc=-2006)
```




Every time the tracking workflow starts, it pulls all target mappings and updates the data based on the configuration.

In this case, one target mapping is causing the issue. Please see below:

![](assets/a06a8deb-6536-ec11-b6e6-000d3a348885.png)

Basically,<b> NMS: tracking log</b> schema should NOT be used for monitoring logs, because the schema is not related to a database (SQL) table.

In conclusion, the issue has nothing to do with the upgrade, but more with a bad target mapping configuration.
